opentracing-noop/src/main/java/io/opentracing/NoopTracer.java

```diff
@@ -15,19 +15,25 @@
 
 import io.opentracing.propagation.Format;
 
-public final class NoopTracer implements Tracer {
+public interface NoopTracer extends Tracer {
+//    static NoopTracer create() {
+//        return NoopTracerImpl.INSTANCE;
+//    }
+}
 
-    public final static NoopTracer INSTANCE = new NoopTracer();
+final class NoopTracerImpl implements NoopTracer {
+    final static NoopTracer INSTANCE = new NoopTracerImpl();
 
     @Override
     public SpanBuilder buildSpan(String operationName) {
-        return NoopSpanBuilder.INSTANCE;
+        return NoopSpanBuilderImpl.INSTANCE;
     }
 
     @Override
     public <C> void inject(SpanContext spanContext, Format<C> format, C carrier) {}
 
     @Override
-    public <C> SpanBuilder extract(Format<C> format, C carrier) { return NoopSpanBuilder.INSTANCE; }
+    public <C> SpanBuilder extract(Format<C> format, C carrier) { return NoopSpanBuilderImpl.INSTANCE; }
```

Comment:
The NoopTracer has no need to return a SpanBuilder instead of a SpanContext.
Have fixed this. Thanks for pointing it out.

> 1. extracting from a request that has no tracing context => return some form of "nothing", which can still be used in childof(), but will be ignored and a new trace is started

I'm not sure I agree with this logic. Calling extract() doesn't result in a request that isn't being sampled becoming sampled. There would be a huge amplification of tracing if that was the case.